Configuration/Applications/python/ConfigBuilder.py

```diff
@@ -1676,6 +1677,17 @@ def prepare_PAT(self, sequence = "miniAOD"):
 
         return
 
+    def prepare_PATGEN(self, sequence = "miniAOD"):
+        ''' Enrich the schedule with PATGEN '''
+        self.loadDefaultOrSpecifiedCFF(sequence,self.PATDefaultCFF,1) #this is unscheduled
+	if not self._options.runUnscheduled:	
+		raise Exception("MiniAOD production can only run in unscheduled mode, please run cmsDriver with --runUnscheduled")
```

Comment:
this should probably be changed to PATGEN to remove confusion with the full proper miniAOD